interviewTest/Task2.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ This is simple Login form component.
+
+ 1. There are 3x fields - a user name (text input), a password (text input) and a 'remember me' checkbox
+ 2. All 3 fields should be editable by the user
+ 3. The form can be submitted by the user with the correct data
+
+ There are issues with this implementation. Can you see them?
+*/
+import CheckBox from "@react-native-community/checkbox";
+import React, { useState } from "react";
+import { Button, StyleSheet, Text, TextInput, View } from "react-native";
+import { isTablet } from "react-native-device-info";
+
+type Props = {
+  onSubmit: () => void;
+};
+
+export const LoginForm = (props: Props) => {
+  const { onSubmit } = props;
+
+  const [userName, setUserName] = useState("");
+  const [password, setPassword] = useState("");
+  const [rememberMe, setRememberMe] = useState(false);
+
+  const onNameChange = (userName: string) => setUserName(userName);
+
+  const onPasswordChange = (password: string) => setPassword(password);
+
+  const onRememberChange = (remember: boolean) => setRememberMe(remember);
```

Comment:
Nit: these change handlers should have been declared with `useCallback`. That way, the app avoid redeclaring them on every render since they ostensibly have no reactive dependencies.
```suggestion
  const onRememberChange = useCallback((remember: boolean) => setRememberMe(remember), []);
```